include/emitc/emitc_tosa.h

```diff
@@ -104,6 +104,78 @@ inline Src sub(Src x, Src y) {
 }
 
 /// Other ops
+// Conv2DOp
+template <typename Dest, typename Src, typename Weights>
+Dest conv2d(Src input, Weights weights, Tensor1D<int64_t, 4> padding,
+            Tensor1D<int64_t, 2> stride, Tensor1D<int64_t, 2> dilation) {
+  // This implementation is taken from emitc_mhlo.c (convolution) and slightly
+  // adapted to fit the memory layout of tosa. Input is [N,IH,IW,IC], weights
+  // are [OC,KH,KW,IC] and output is [N,H,W,OC].
+  static_assert(is_tensor_of_dim<4, Src>::value,
+                "Expected 4 dimensional input");
+  static_assert(is_tensor_of_dim<4, Dest>::value,
+                "Expected 4 dimensional output");
+  static_assert(is_tensor_of_dim<4, Weights>::value,
+                "Expected 4 dimensional weights");
+
+  assert(stride[0] == 1);
+  assert(stride[0] == 1);
```

Comment:
`stride[1] == 1`

But why do you limit stride to be 1? Looking at the implementation strides are in fact supported.

Reply:
I took this from `emitc_mhlo.h` where the same checks are applied. Also we don't have tests for this. So it is not really supported and not tested. That's why there are this checks. Since we are not aiming for full feature compatibility in our first shot, I would leave it as it is.